microsoft/testsuites/core/provisioning.py

```diff
@@ -300,6 +300,55 @@ def stress_reboot(self, log: Logger, node: RemoteNode, log_path: Path) -> None:
             log.info(f"Average reboot time: {mean(times):.2f}s")
             log.info(f"Median reboot time: {median(times):.2f}s")
 
+    @TestCaseMetadata(
+        description="""
+        This test case verifies that the system can boot and provision successfully
+        with the swiotlb=force kernel parameter enabled. This kernel parameter
+        forces the use of software I/O TLB for all DMA operations.
+
+        This is particularly relevant in Confidential Computing VMs,
+        where memory is encrypted and direct DMA access isn't possible.
+        In such cases, bounce buffering becomes mandatory.
+
+        Steps:
+        1. Set the swiotlb=force kernel parameter in grub configuration
+        2. Reboot the system to apply the kernel parameter
+        3. Run smoke test to verify system functionality
+        4. Verify the system is responsive after reboot
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_features=[SerialConsole, StartStop],
+        ),
+    )
+    def verify_deployment_provision_swiotlb_force(
+        self, log: Logger, node: RemoteNode, log_path: Path
+    ) -> None:
+        # Check if CONFIG_SWIOTLB is available in kernel configuration
+        kernel_config = node.tools[KernelConfig]
+        if not kernel_config.is_enabled("CONFIG_SWIOTLB"):
+            raise SkippedException("CONFIG_SWIOTLB is not enabled in kernel")
+
+        grub_config = node.tools[GrubConfig]
+        grub_config.set_kernel_cmdline_arg("swiotlb", "force")
+        node.features[StartStop].restart(wait=True)
+
+        cmdline_result = node.execute("cat /proc/cmdline", sudo=True)
```

Comment:
use cat tool

Reply:
overlooked this one, thanks for pointing out. Fixed